contracts/mock/ProtocolInitializationTestFacet.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.9;
+
+import "../domain/BosonConstants.sol";
+import { IBosonProtocolInitializationHandler } from "../interfaces/handlers/IBosonProtocolInitializationHandler.sol";
+import { ProtocolLib } from "../protocol/libs/ProtocolLib.sol";
+import { ProtocolBase } from "../protocol/bases/ProtocolBase.sol";
+import { DiamondLib } from "../diamond/DiamondLib.sol";
+
+/**
+ * @title IBosonProtocolInitializationTestHandler
+ *
+ * @notice Mock changes to IBosonProtocolInitializationFacet.
+ *
+ */
+contract ProtocolInitializationTestFacet is IBosonProtocolInitializationHandler, ProtocolBase {
+    /**
+     * @notice Modifier to protect initializer function from being invoked twice for a given version.
+     */
+    modifier onlyUnInitializedVersion(bytes32 _version) {
+        ProtocolLib.ProtocolStatus storage ps = protocolStatus();
+        require(!ps.initializedVersions[_version], ALREADY_INITIALIZED);
+        ps.initializedVersions[_version] = true;
+        _;
+    }
+
+    /**
+     * @notice Initializes the protocol after the deployment.
+     * This function is callable only once for each version
+     *
+     * @param _version - version of the protocol
+     * @param _addresses - array of facet addresses to call initialize methods
+     * @param _calldata -  array of facets initialize methods encoded as calldata
+     *                    _calldata order must match _addresses order
+     * @param _isUpgrade - flag to indicate whether this is first deployment or upgrade
+     * @param interfacesToRemove - array of interfaces to remove from the diamond
+     * @param interfacesToAdd - array of interfaces to add to the diamond
+     *
+     */
+    function initialize(
+        bytes32 _version,
+        address[] calldata _addresses,
+        bytes[] calldata _calldata,
+        bool _isUpgrade,
+        bytes4[] calldata interfacesToRemove,
+        bytes4[] calldata interfacesToAdd
+    ) external onlyUnInitializedVersion(_version) {
+        require(_version != bytes32(0), VERSION_MUST_BE_SET);
+        require(_addresses.length == _calldata.length, ADDRESSES_AND_CALLDATA_LENGTH_MISMATCH);
+
+        // Delegate call to initialize methods of facets declared in _addresses
+        for (uint256 i = 0; i < _addresses.length; i++) {
+            (bool success, bytes memory error) = _addresses[i].delegatecall(_calldata[i]);
+
+            // Handle result
+            if (!success) {
+                if (error.length > 0) {
+                    // bubble up the error
+                    revert(string(error));
+                } else {
+                    revert(PROTOCOL_INITIALIZATION_FAILED);
+                }
+            }
+        }
+
+        if (_isUpgrade) {
+            if (_version == bytes32("2.2.0")) {
+                initV2_2_0();
+            } else if (keccak256(abi.encodePacked(_version)) == keccak256(abi.encodePacked(bytes32(bytes("2.2.1"))))) {}
```

Comment:
i don't understand this, why bother with the `else if` if there is no action ?

Reply:
I just realised that this is in a mock ...

Reply:
yes, I had created this mock to use on unit tests